pkg/network/ebpf/c/protocols/events.h

```diff
@@ -54,8 +55,19 @@
                 }                                                                                       \
                                                                                                         \
                 if (use_ring_buffer) {                                                                  \
-                    perf_ret = bpf_ringbuf_output(&name##_batch_events, batch, sizeof(batch_data_t), 0);\
+                    if (with_telemetry) {                                                               \
+                        perf_ret = bpf_ringbuf_output_with_telemetry(&name##_batch_events, batch, sizeof(batch_data_t), 0);\
+                    } else {                                                                            \
+                        perf_ret = bpf_ringbuf_output(&name##_batch_events, batch, sizeof(batch_data_t), 0);\
+                    }                                                                                   \
                 } else {                                                                                \
+                    if (with_telemetry) {                                                               \
+                        perf_ret = bpf_perf_event_output_with_telemetry(ctx,                            \
+                                                         &name##_batch_events,                          \
+                                                         key.cpu,                                       \
+                                                         batch,                                         \
+                                                         sizeof(batch_data_t));                         \
+                    }                                                                                   \
                     perf_ret = bpf_perf_event_output(ctx,                                               \
```

Comment:
Take a look in `READ_INTO_USER_BUFFER_INTERNAL`, `READ_INTO_USER_BUFFER` and `READ_INTO_USER_BUFFER_WITHOUT_TELEMETRY`

Try to see if you can move the output function as an argument to the define and simplify the method

Reply:
Fixed

Reply:
On a second thought, using the original appraoch is better as it let's the compiler the function based on the with_telemetry flag. In this approach, the function will always be duplicated, increasing the program size.